test/export/test_export.py

```diff
@@ -6611,13 +6611,20 @@ def forward(self, q, k, v):
         ep = torch.export.export(
             ScaledDotProductAttention(), (q, k, v)
         ).run_decompositions()
-        self.assertExpectedInline(
-            ep.graph_module.code.strip(),
-            """\
+        code_str = """\
 def forward(self, q, k, v):
     _scaled_dot_product_flash_attention = torch.ops.aten._scaled_dot_product_flash_attention.default(q, k, v, 0.0, True, scale = 0.125);  q = k = v = None
     getitem = _scaled_dot_product_flash_attention[0];  _scaled_dot_product_flash_attention = None
-    return (getitem,)""",
+    return (getitem,)"""
+        if SM90OrLater:
+            code_str = """\
+def forward(self, q, k, v):
+    _scaled_dot_product_cudnn_attention = torch.ops.aten._scaled_dot_product_cudnn_attention.default(q, k, v, None, False, 0.0, True);  q = k = v = None
+    getitem = _scaled_dot_product_cudnn_attention[0];  _scaled_dot_product_cudnn_attention = None
```

Comment:
So the difference is that since we pad out the head_dim for flashattention we actually pass it into the op:

https://github.com/pytorch/pytorch/blob/b40249b462702e39610398e69b2f5b88744012bb/aten/src/ATen/native/transformers/attention.cpp#L674

versus cudnn:
https://github.com/pytorch/pytorch/blob/b40249b462702e39610398e69b2f5b88744012bb/aten/src/ATen/native/transformers/attention.cpp#L662